tests/unit/metrics/StatsClient.js

```diff
@@ -27,52 +28,110 @@ describe('StatsClient class', () => {
+                    assert.deepEqual(res, expected);
+                    next();
+                });
+            },
+        ], done);
+    });
 
-            const expected = [[null, 2], [null, 1]];
-            assert.deepEqual(res, expected);
-        });
+    it('should record new requests by defined amount increments', done => {
+        function noop() {}
+
+        async.series([
+            next => {
+                statsClient.reportNewRequest(id, 9);
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+
+                    assert.equal(res.requests, 9);
+                    next();
+                });
+            },
+            next => {
+                statsClient.reportNewRequest(id);
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+
+                    assert.equal(res.requests, 10);
+                    next();
+                });
+            },
+            next => {
+                statsClient.reportNewRequest(id, noop);
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+
+                    assert.equal(res.requests, 11);
+                    next();
+                });
+            },
+        ], done);
     });
 
-    it('should correctly record a 500 on the server', () => {
+    it('should correctly record a 500 on the server', done => {
         statsClient.report500(id, (err, res) => {
             assert.ifError(err);
             assert(Array.isArray(res));
             assert.equal(res.length, 2);
 
             const expected = [[null, 1], [null, 1]];
             assert.deepEqual(res, expected);
+            done();
         });
     });
 
-    it('should respond back with total requests', () => {
-        statsClient.reportNewRequest(id, err => {
-            assert.ifError(err);
-        });
-        statsClient.report500(id, err => {
-            assert.ifError(err);
-        });
-        statsClient.getStats(fakeLogger, id, (err, res) => {
-            assert.ifError(err);
-            assert.equal(typeof res, 'object');
-            assert.equal(Object.keys(res).length, 3);
-            assert.equal(res.sampleDuration, STATS_EXPIRY);
+    it('should respond back with total requests', done => {
+        async.series([
+            next => {
+                statsClient.reportNewRequest(id, err => {
+                    assert.ifError(err);
+                    next();
+                });
+            },
+            next => {
+                statsClient.report500(id, err => {
+                    assert.ifError(err);
+                    next();
+                });
+            },
+            next => {
+                statsClient.getStats(fakeLogger, id, (err, res) => {
+                    assert.ifError(err);
+                    assert.equal(typeof res, 'object');
```

Comment:
You could get rid of the following lines since you're checking the whole object just next. (for debugging failed tests, no worry since mocha will dump the obtained and expected objects when it fails, so we know exactly what has been returned in the callback).

I also saw a couple other instances above where using `deepEqual` makes the other tests redundant.